Tests/MapboxNavigationTests/MapViewTests.swift

```diff
@@ -137,4 +138,181 @@ class MapViewTests: XCTestCase {
+            
+            var expression: Exp? = nil
+            XCTAssertNoThrow(expression = try JSONDecoder().decode(Exp.self, from: expressionData!))
+            return expression
+        }
+        
+        XCTAssertEqual(textFieldExpression(layerIdentifier: "road-labels"),
+                       Exp(.format) {
+                        Exp(.coalesce) { Exp(.get) { "name_en" }; Exp(.get) { "name" } }
+                        FormatOptions()
+                       },
+                       "Road labels should be in English by default.")
+        XCTAssertEqual(textFieldExpression(layerIdentifier: "place-labels"),
+                       Exp(.format) {
+                        Exp(.coalesce) { Exp(.get) { "name_en" }; Exp(.get) { "name" } }
+                        FormatOptions()
+                       },
+                       "Place labels should be in English by default.")
+        
+        mapView.localizeLabels(into: Locale(identifier: "en"))
+        
+        // TODO: Unlocalize road labels: https://github.com/mapbox/mapbox-maps-ios/issues/653
+        XCTAssertEqual(textFieldExpression(layerIdentifier: "road-labels"),
+                       Exp(.format) {
+                        Exp(.coalesce) { Exp(.get) { "name_en" }; Exp(.get) { "name" } }
+                        FormatOptions()
+                       },
+                       "Road labels should remain in English after localization.")
+        XCTAssertEqual(textFieldExpression(layerIdentifier: "place-labels"),
+                       Exp(.format) {
+                        Exp(.coalesce) { Exp(.get) { "name_en" }; Exp(.get) { "name" } }
+                        FormatOptions()
+                       },
+                       "Place labels should be in English after localization.")
+        
+        mapView.localizeLabels(into: Locale(identifier: "es"))
+        
+        XCTAssertEqual(textFieldExpression(layerIdentifier: "road-labels"),
+                       Exp(.format) {
+                        Exp(.coalesce) { Exp(.get) { "name_en" }; Exp(.get) { "name" } }
+                        FormatOptions()
+                       },
+                       "Road labels should remain in English after localization.")
+        XCTAssertEqual(textFieldExpression(layerIdentifier: "place-labels"),
+                       Exp(.format) {
+                        Exp(.coalesce) { Exp(.get) { "name_es" }; Exp(.get) { "name" } }
+                        FormatOptions()
+                       },
+                       "Place labels should be in Spanish after localization.")
+        
+        mapView.localizeLabels(into: Locale(identifier: "zh-Hant-TW"))
+        
+        XCTAssertEqual(textFieldExpression(layerIdentifier: "road-labels"),
+                       Exp(.format) {
+                        Exp(.coalesce) { Exp(.get) { "name_en" }; Exp(.get) { "name" } }
+                        FormatOptions()
+                       },
```

Comment:
This test looks pretty large, would it be possible to expose this part into reusable component?

Reply:
I condensed the tests a bit to make them a bit easier to read.